torch/utils/cpp_extension.py

```diff
@@ -250,3 +330,38 @@ def _write_ninja_file(path, name, sources, extra_cflags, extra_ldflags,
         for block in blocks:
             lines = '\n'.join(block)
             build_file.write('{}\n\n'.format(lines))
+
+
+def _wrap_compile(obj, src, ext, cc_args, extra_postargs, pp_opts):
+    # Recover the original compiler instance that we are monkey-patching.
+    self = _wrap_compile.compiler_object
+
+    # Store the original compiler for later.
+    original_compiler = self.compiler_so
+    if os.path.splitext(src)[1].startswith('.cu'):
+        self.set_executable('compiler_so', _join_cuda_home('bin', 'nvcc'))
+        if isinstance(extra_postargs, dict):
+            extra_postargs = extra_postargs['nvcc']
+        extra_postargs += ['-c', '--compiler-options', "'-fPIC'"]
+    else:
+        if isinstance(extra_postargs, dict):
+            extra_postargs = extra_postargs['cxx']
+        extra_postargs.append('-std=c++11')
+
+    self._original_compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
+
+    # Put the original compiler back in place.
+    self.set_executable('compiler_so', original_compiler)
```

Comment:
This should probably be implemented as a `try: ... finally: ...`

Reply:
Ok 👍